src/components/RichContenteditable/RichContenteditable.vue

```diff
@@ -0,0 +1,301 @@
+
+### Examples
+
+```vue
+<template>
+	<div>
+		<RichContenteditable
+			v-model="message"
+			:auto-complete="autoComplete"
+			placeholder="Try mentioning the user Test01"
+			style="width: 350px;height: 100px;" />
+		{{ message }}
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			message: '',
+		}
+	},
+	methods: {
+		/**
+		* Do your own query to the autocompletion api.
+		* The returned data bellow is a fake data example.
+		* The callback expects the same format returned by the core/autocomplete/get ocs api endpoint!
+		* @param {string} search the query
+		* @param {Function} callback the callback to process the results with
+		*/
+		autoComplete(search, callback) {
+			return callback([{
+				icon: 'icon-user',
+				id: 'Test01',
+				label: 'Test01',
+				source: 'users',
+			}])
+		}
+	}
+}
+</script>
+```
+
+</docs>
+
+<template>
+	<InputContenteditable
+		_is="div"
+		:value="value"
+		:placeholder="placeholder"
+		class="rich-contenteditable__input"
+		@input="onInput" />
+</template>
+
+<script>
+import InputContenteditable from 'vue-input-contenteditable'
+
+import Tribute from 'tributejs/dist/tribute.esm'
+import debounce from 'debounce'
+import { generateUrl } from '@nextcloud/router'
+
+import { t } from '../../l10n.js'
+
+export default {
+	name: 'RichContenteditable',
+
+	components: {
+		InputContenteditable,
+	},
+
+	props: {
+		value: {
+			type: String,
+			required: true,
+		},
+		placeholder: {
+			type: String,
+			default: t('Write message, @ to mention someone …'),
+		},
+		autoComplete: {
+			type: Function,
+			required: true,
+		},
+		menuContainer: {
+			type: Element,
+			default: () => document.body,
+		},
+	},
+
+	data() {
+		return {
+			tribute: null,
+			options: {
+				fillAttr: 'id',
+				// Search against id and label (display name)
+				lookup: result => `${result.id} ${result.label}`,
+				menuContainer: this.menuContainer,
+				menuItemTemplate: this.genTemplate,
+				noMatchTemplate: () => '<span class="hidden"></span>',
+				selectTemplate: item => `@${item.original.id}`,
+				values: this.debouncedAutoComplete,
+			},
+		}
+	},
+
+	mounted() {
+		this.tribute = new Tribute(this.options)
+		this.tribute.attach(this.$el)
+	},
+	beforeDestroy() {
+		if (this.tribute) {
+			this.tribute.detach(this.$el)
+		}
+	},
+
+	methods: {
+		/**
+		 * Re-emit the input event to the parent
+		 * @param {string} text the message
+		 */
+		onInput(text) {
+			this.$emit('input', text)
+			this.$emit('update:value', text)
+		},
+
+		/**
+		 * Debounce the autocomplete function
+		 */
+		debouncedAutoComplete: debounce(async function(search, callback) {
+			this.autoComplete(search, callback)
+		}, 200),
+
+		genTemplate({ original }) {
+			const avatarUrl = original.source === 'users'
+				? this.getAvatarUrl(original.id)
+				: null
+
+			const haveStatus = original.status?.icon || original.status?.status
+
+			return `<div class="autocomplete-result" data-v-${SCOPE_VERSION}>
```

Comment:
:eye: :lips: :eye: 

looks like old Backbone+Handlebars code

Reply:
> :eye: :lips: :eye: 

Man I love this emoji combo :D
Yep, that's the only issue with this component.
Though tiny simple div rendering. Good enough to you?

I thought there might be a way to generate a Vue component and get the html directly from it, but couldn't do it. An unmounted component doesn't return the element (make sense), nor the html.

Reply:
Ah found a way! Rending is fast too!
```
Rendering took 3.540000005159527 ms
Rendering took 1.0350000229664147 ms
Rendering took 1.0450000409036875 ms
Rendering took 1.0550000006332994 ms
```